va/va.c

```diff
@@ -1679,6 +1679,25 @@ VAStatus vaQuerySurfaceError (
   return va_status;
 }
 
+VAStatus vaSyncBuffer (
+    VADisplay dpy,
+    VABufferID buf_id,
+    uint64_t timeout_ns
+)
+{
+  VAStatus va_status;
+  VADriverContextP ctx;
+
+  CHECK_DISPLAY(dpy);
+  ctx = CTX(dpy);
+
+  va_status = ctx->vtable->vaSyncBuffer( ctx, buf_id, timeout_ns );
```

Comment:
It is possible the underlying driver doesn't implement this callback function, so we should check ctx->vtabl->vaSyncBuffer against NULL before calling it. 
```suggestion
  if (ctx->vtable->vaSyncBuffer)
       va_status = ctx->vtable->vaSyncBuffer(ctx, buf_id, timeout_ns);
  else
       va_status = VA_STATUS_ERROR_UNIMPLEMENTED;
```

Reply:
Thanks! Done.